modules/images/image-loading-optimization/detection.php

```diff
@@ -45,7 +48,9 @@ function ilo_get_detection_script( string $slug, array $needed_minimum_viewport_
 		'urlMetricsNonce'             => ilo_get_url_metrics_storage_nonce( $slug ),
 		'neededMinimumViewportWidths' => $needed_minimum_viewport_widths,
 		'storageLockTTL'              => ilo_get_url_metric_storage_lock_ttl(),
+		'webVitalsLibrarySrc'         => esc_url( $web_vitals_lib_src ),
```

Comment:
Actually, this should not be escaped. This is not where it is rendered. It will be escaped by `json_encode()` below. At most it could be sanitized with `esc_url_raw()` but I think this would be overkill since it is essentially a string literal defined above.

```suggestion
		'webVitalsLibrarySrc'         => $web_vitals_lib_src,
```

Importantly, `esc_url()` is for _HTML attribute_ escaping, so it will output `&` as `&amp;` which will not work properly in a script context.